pkg/strategy/dca2/strategy.go

```diff
@@ -289,30 +291,29 @@ func (s *Strategy) CleanUp(ctx context.Context) error {
 		return fmt.Errorf("Session is nil, please check it")
 	}
 
-	service, support := session.Exchange.(advancedOrderCancelApi)
-	if !support {
-		return fmt.Errorf("advancedOrderCancelApi interface is not implemented, fallback to default graceful cancel, exchange %T", session)
+	// ignore the first cancel error, this skips one open-orders query request
+	if err := tradingutil.UniversalCancelAllOrders(ctx, session.Exchange, nil); err == nil {
+		return nil
 	}
 
+	// if cancel all orders returns error, get the open orders and retry the cancel in each round
 	var werr error
 	for {
 		s.logger.Infof("checking %s open orders...", s.Symbol)
 
 		openOrders, err := retry.QueryOpenOrdersUntilSuccessful(ctx, session.Exchange, s.Symbol)
 		if err != nil {
-			s.logger.WithError(err).Errorf("CancelOrdersByGroupID api call error")
-			werr = multierr.Append(werr, err)
+			s.logger.WithError(err).Errorf("unable to query open orders")
+			continue
```

Comment:
Will this hit api rate limit?

Reply:
I think we have the protection in the exchange api layer